watch/funcs.go

```diff
@@ -41,7 +42,7 @@ func keyWatch(params map[string]interface{}) (WatcherFunc, error) {
 	}
 	fn := func(p *Plan) (uint64, interface{}, error) {
 		kv := p.client.KV()
-		opts := consulapi.QueryOptions{AllowStale: stale, WaitIndex: p.lastIndex}
+		opts := makeQueryOptionsWithContext(p, stale)
```

Comment:
Do all of these watch functions need a `defer cancel()` in here to clean up in case the request goes through?